diff --git a/dkms.conf b/dkms.conf
new file mode 100644
index 0000000..a5f0e44
--- /dev/null
+++ b/dkms.conf
@@ -0,0 +1,9 @@
+MAKE="make -C product/rr272x/linux/ KERNELDIR=/lib/modules/${kernelver}/build"
+CLEAN="make -C product/rr272x/linux/ clean"
+BUILT_MODULE_NAME=rr272x
+DEST_MODULE_LOCATION=/kernel/drivers/scsi/
+BUILT_MODULE_LOCATION=product/rr272x/linux/
+PACKAGE_NAME=rr272x
+PACKAGE_VERSION=1.5
+AUTOINSTALL=yes
+REMAKE_INITRD=yes
