--- a/inc/linux_64mpa/Makefile.def
+++ b/inc/linux_64mpa/Makefile.def
@@ -93,10 +93,12 @@
 $(error Cannot find kernel version. Check $(KERNELDIR)/include/linux/version.h.)
 endif
 
+ifneq ($(MAJOR), 4)
 ifneq ($(MAJOR), 3)
 ifneq ($(KERNEL_VER), 2.6)
 ifneq ($(KERNEL_VER), 2.4)
-$(error Only kernel 2.4/2.6/3.x is supported but you use $(KERNEL_VER))
+$(error Only kernel 2.4/2.6/3.x/4.x is supported but you use $(KERNEL_VER))
+endif
 endif
 endif
 endif
--- a/osm/linux/install.sh
+++ b/osm/linux/install.sh
@@ -10,7 +10,7 @@
 	OBJ=o
 	MODVER=`modinfo -f%{kernel_version} ${PWD}/${TARGETNAME}.${OBJ}`
 	;;
-	2.6 | 3.* )
+	2.6 | 3.* | 4.* )
 	OBJ=ko
 	MODVER=`modinfo -F vermagic ${PWD}/${TARGETNAME}.${OBJ} | cut -d' ' -f1`
 	;;
--- a/product/rr272x/linux/config.c
+++ b/product/rr272x/linux/config.c
@@ -24,7 +24,7 @@
 
 char driver_name[] = "rr272x_1x";
 char driver_name_long[] = "RocketRAID 272x_1x controller driver";
-char driver_ver[] = "v1.6.1 (" __DATE__ " " __TIME__ ")";
+char driver_ver[] = "v1.6.1";
 int  osm_max_targets = 32;
 int os_max_cache_size = 0x2000000;
 
